p4-16/psa/PSA.mdk

```diff
@@ -1748,3 +1748,357 @@ types, as the P4 program author wishes.
+  caption: Result of packet processed one time by ingress or egress.}
+|--------------|-------------|---------------|--------------------|
+|              |             | Processed     | Resulting          |
+| Abbreviation | Description | next by       | packet(s)          |
++:-------------+:------------+:--------------+:-------------------+
+| NFP          | normal packet from port |         | At most one CI2E |
+|--------------|-------------|           |                    |
+| NFCPU        | packet from CPU port    | ingress | packet, plus at most |
+|--------------|-------------|           |                    |
+| RESUB        | resubmitted packet      |         | one of a RESUB,  |
+|--------------|-------------|           |                    |
+| RECIRC       | recirculated packet     |         | NU, or NM packet |
+|--------------|-------------|-----------|--------------------|
+| NU           | normal unicast packet       |         | At most one CE2E     |
+|              | from ingress to egress      |         | packet, plus at most |
+|--------------|-------------|               |                    |
+| NM           | normal multicast-replicated | egress  | one of a RECIRC,     |
+|              | packet from ingress to egress |       | NTP, or NTCPU packet |
+|--------------|-------------|               |                    |
+| CI2E         | clone from ingress to egress |        |          |
+|--------------|-------------|               |                    |
+| CE2E         | clone from egress to egress |         |          |
+|--------------|-------------|---------------|--------------------|
+| NTP          | normal packet to port   | device at other | determined by the |
+|              |                         | end of port     | other device      |
+|--------------|-------------|---------------|--------------------|
+| NTCPU        | normal packet to CPU port | CPU | determined by CPU |
+|--------------|-------------|---------------|--------------------|
+~
+
+For egress packets, the choice of egress port, or the port to the CPU,
+is made by the immediately previous processing step (ingress for NU,
+NM, or CI2E packets, egress for CE2E packets), and egress processing
+cannot change that choice to a different port.  It can change the
+packet to RECIRC instead, or drop the packet.
+
+A single packet received by a PSA system from a port can result in 0,
+1, or many packets going out, all under control of the PSA program.
+For example, a single packet received from a port could cause all of
+the following to occur, if the PSA program so directed it:
+
++ The original packet received as NFP from port 2.  Ingress processing
+  creates a CI2E clone destined for the CPU port (copy 1), and a
+  multicast NM packet to multicast group 18, which is configured in
+  the PacketReplicationEngine to have copies made to ports 5 (copy 2)
+  and 7 (copy 3).
++ Copy 1 performs egress processing, which does not create a CE2E
```

Comment:
Readability: do we need to mention what it does not do? if so, i would move it after explaining what the copy does. E.g.: Copy 1 performs egress processing and sends the packet on path NTCPU to the CPU port. It does not create a CE2E clone.

Reply:
I suppose we could leave out the "does not make a clone" phrases here, with the assumption that none is made if it doesn't say anything about a clone.

Reply:
Done.